tsinfer/inference.py

```diff
@@ -2285,7 +2285,9 @@ def get_samples_tree_sequence(self, map_additional_sites=True):
 
         # All true ancestors are samples in the ancestors tree sequence. We unset
         # the SAMPLE flag but keep other flags intact.
-        new_flags = np.bitwise_and(tables.nodes.flags, ~tskit.NODE_IS_SAMPLE)
+        new_flags = np.bitwise_and(
```

Comment:
We're taking a whole other copy of the flags column here just to get the dtype. Can't we just case to uint32 explicitly? That's never going to change.

Reply:
Oh, good point. I would slightly prefer simply to make a variable pointing at the flags, rather than hardcoding types. Does that work?

```
node_flags = tables.nodes.flags
new_flags = np.bitwise_and(nodes_flags, ~nodes_flags.dtype.type(tskit.NODE_IS_SAMPLE))
```